libcxx/include/__exception/exception_ptr.h

```diff
@@ -36,7 +36,12 @@ struct __cxa_exception;
 _LIBCPP_OVERRIDABLE_FUNC_VIS __cxa_exception* __cxa_init_primary_exception(
     void*,
     std::type_info*,
-    void(
+#  if defined(__USING_WASM_EXCEPTIONS__)
```

Comment:
Let's introduce a typedef name for this function pointer type. Then we can do something like

```c++
#ifdef __USING_WASM_EXCEPTIONS__
using _ExceptionDtor = something;
#else
using _ExceptionDtor = something-else;
#endif
```

and that should be more readable than the current approach.

Reply:
yeah. but here's a problem. __USING_WASM_EXCEPTIONS__ is not defined by clang 19 even with -fwasm-exceptions enabled. i suggest to just use __wasi__ macro, instead of __USING_WASM_EXCEPTIONS__

Reply:
ok i will try to fix it.

Reply:
Fixed